src/helm/benchmark/metrics/language_modeling_metrics.py

```diff
@@ -0,0 +1,106 @@
+from collections import defaultdict
+from typing import List, Dict, Set
+
+from helm.benchmark.adaptation.scenario_state import ScenarioState
+from helm.benchmark.metrics.basic_metrics import (
+    compute_language_modeling_metrics,
+    compute_perplexity_metrics,
+    compute_request_state_metrics,
+)
+from helm.benchmark.metrics.evaluate_reference_metrics import compute_reference_metrics
+from helm.benchmark.metrics.efficiency_metrics import EfficiencyMetric
+
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.adaptation.adapter_spec import AdapterSpec
+from .metric import MetricInterface, MetricResult, PerInstanceStats, add_context
+from .metric_name import MetricContext, MetricName
+from .metric_service import MetricService
+from .statistic import Stat, merge_stat
+
+
+class LanguageModelingMetric(MetricInterface):
+    """
+    Defines the basic metrics available when using the ADAPT_LANGUAGE_MODELING adapter.
+    This is parallel to BasicMetric and produces many of the same Stats.
+    """
+
+    def __init__(self, names: List[str]):
+        self.names: List[str] = names
+        self.efficiency_metric = EfficiencyMetric()
+
+    def __repr__(self):
+        # TODO: Should this be changed to LanguageModelingMetric?
```

Comment:
Yes, this should be `LanguageModelingMetric` - it should not affect the generated JSON.

Reply:
Done.